<a href="https://colab.research.google.com/github/CristianLLDev/talentotech/blob/main/Funciones_calculo_MAE%2CMSE%2CR2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

FUNCIONES PARA CALCULAR Error Cuadrático Medio (MSE), Error Absoluto Medio (MAE), Coeficiente de Determinación (R²)

In [1]:
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score

def get_regression_metrics(y_real, y_pred):
    # Error Absoluto Medio (MAE)
    mae = mean_absolute_error(y_real, y_pred)

    # Error Cuadrático Medio (MSE)
    mse = mean_squared_error(y_real, y_pred)

    # Coeficiente de Determinación (R²)
    r2 = r2_score(y_real, y_pred)

    return {"MAE": mae, "MSE": mse, "R²": r2}

# Ejemplo de uso
y_real = [3.0, -0.5, 2.0, 7.0]
y_pred = [2.5, 0.0, 2.0, 8.0]

metrics = get_regression_metrics(y_real, y_pred)
print(metrics)


{'MAE': 0.5, 'MSE': 0.375, 'R²': 0.9486081370449679}


mean_absolute_error(y_real, y_pred): r/ 0.5 Calcula el error absoluto medio, que es el promedio de los valores absolutos de las diferencias entre y_real y y_pred.

mean_squared_error(y_real, y_pred):r/ 0.375 Calcula el error cuadrático medio, que es el promedio de las diferencias al cuadrado entre y_real y y_pred.

r2_score(y_real, y_pred): r/ 'R²': 0.9486081370449679  Calcula el coeficiente de determinación (R²), que mide qué tan bien se ajusta el modelo a los datos.